tests/backup/backup_upgrade_test.go

```diff
@@ -393,23 +393,25 @@ var _ = Describe("{PXBackupEndToEndBackupAndRestoreWithUpgrade}", func() {
 				backupName := fmt.Sprintf("%s-%s-%v-with-rules", BackupNamePrefix, namespace, time.Now().Unix())
 				labelSelectors := make(map[string]string, 0)
 				log.InfoD("Creating a backup of namespace [%s] with pre and post exec rules", namespace)
-				err := CreateBackup(backupName, SourceClusterName, backupLocationName, backupLocationUid, []string{namespace},
-					labelSelectors, orgID, srcClusterUid, preRuleNames[appName], preRuleUids[appName], postRuleNames[appName], postRuleUids[appName], ctx)
+				appContextsToBackup := FilterAppContextsByNamespace(srcClusterContexts, []string{namespace})
+				err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, backupLocationName, backupLocationUid, appContextsToBackup, labelSelectors, orgID, srcClusterUid, preRuleNames[appName], preRuleUids[appName], postRuleNames[appName], postRuleUids[appName])
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of backup [%s]", backupName))
 				err = IsFullBackup(backupName, orgID, ctx)
 				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if backup [%s] is a full backup", backupName))
 				backupWithRuleNames = SafeAppend(&mutex, backupWithRuleNames, backupName).([]string)
+				backupToContextMapping[backupName] = appContextsToBackup
```

Comment:
How about using a mutex to be safe?

An anonymous function

```
updatebackupToContextMapping := func(backupName string, appContextsToBackup string) {
	mutex.Lock()
	defer mutex.Unlock()
	backupMapping[backupName] = appContextsToBackup
}
```

and

```
updatebackupToContextMapping(backupName, appContextsToBackup)
```

Reply:
done